lib/ood_core/job/adapters/kubernetes/batch.rb

```diff
@@ -308,22 +310,29 @@ def pod_info_from_json(pod)
   end
 
   def configure_auth(auth)
-    type = auth.fetch(:type)
-    return if managed?(type)
-
-    case type
-    when 'gke'
-      set_gke_config(auth)
-    when 'oidc'
-      set_context
-    end
+   if managed_auth?(auth)
+    return
+   elsif gke_auth?(auth)
+    set_gke_config(auth)
+   elsif oidc_auth?(auth)
+    set_context
+   end
   end
 
   def context?
     !@context.nil?
   end
 
-  def managed?(type)
+  def gke_auth?(auth = {})
+    auth.fetch(:type, nil) == 'gke'
+  end
+
+  def oidc_auth?(auth = {})
+    auth.fetch(:type, nil) == 'oidc'
+  end
+
+  def managed_auth?(auth = {})
+    type = auth.fetch(:type, nil)
```

Comment:
Does this say if the auth type isn't set, make it nil, and if it's nil, to return true for a managed auth? Because it seems like if nil is returned then there is no managed auth.